app/javascript/githubRepos/githubRepos.jsx

```diff
@@ -0,0 +1,60 @@
+import { h, Component } from 'preact';
+import { SingleRepo } from './singleRepo';
+
+export class GithubRepos extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      repos: [],
+      erroredOut: false,
+    };
+  }
+
+  componentDidMount() {
+    this.getGithubRepos();
+  }
+
+  getGithubRepos = () => {
+    fetch(`/api/github_repos`, {
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json',
+      },
+      credentials: 'same-origin',
+    })
+      .then(response => response.json())
+      .then(json => {
+        this.setState({ repos: json });
+      })
+      .catch(error => {
+        console.log(error);
+        this.setState({ erroredOut: true });
+      });
+  };
+
+  render() {
+    const { repos, erroredOut } = this.state;
+    const allRepos = repos.map(repo => (
+      <SingleRepo
+        githubIdCode={repo.github_id_code}
+        name={repo.name}
+        fork={repo.fork}
+        selected={repo.selected}
+      />
+    ));
+
+    if (erroredOut) {
+      return (
+        <div className="github-repos github-repos-errored">
+          An error occurred. Please check your browser console and email
+          <a href="mailto:yo@dev.to"> yo@dev.to </a>
+          for more help.
+        </div>
+      );
+    }
+    if (repos.length > 0) {
+      return <div className="github-repos">{allRepos}</div>;
+    }
+    return <div className="github-repos loading-repos" />;
+  }
+}
```

Comment:
Also, consider adding a static `displayName` to your components so that you can debug them more easily in the React devtools. And if you want them to be debuggable in the React devtools, see an example of how to do that here, https://github.com/thepracticaldev/dev.to/commit/18980034d3f006164065980d8a491be7408160e6#diff-8df115e9b1d4f0c7af7646b6dd6e974cR1

Having said that, I actually forgot to add `displayName` to the Search components: 🙃

Reply:
Hmm not sure how to add the `displayName`. Is it as simple as changing the div to `<div className="..." displayName="githubRepos">`?

Reply:
The [displayName](https://reactjs.org/docs/react-component.html#displayname) is a static property that you set on your Preact component. You generally set it after you create the class or function for tou  component. e.g.

```
GithubRepos.displayName = 'GitHubRepos'
```

Reply:
The [displayName](https://reactjs.org/docs/react-component.html#displayname) is a static property that you set on your Preact component. You generally set it after you create the class or function for tou  component. e.g.

```
GithubRepos.displayName = 'GitHubRepos'
```